kimchi/src/tests/foreign_field_add.rs

```diff
@@ -844,6 +842,38 @@ fn test_random_small_sub() {
     );
 }
 
+#[test]
+// Test with bad parameters in bound check
+fn test_bad_bound() {
+    let foreign_mod = BigUint::from_bytes_be(&SECP256K1_MOD);
+    let left_input = random_input(foreign_mod.clone(), false);
+    let right_input = random_input(foreign_mod.clone(), false);
+    let (mut witness, cs) = test_ffadd(
+        SECP256K1_MOD,
+        vec![left_input.clone(), right_input.clone()],
+        &vec![FFOps::Add],
+    );
+    // Modify sign of bound
+    // It should be constrained that sign needs to be 1
+    witness[6][17] = -PallasField::one();
+    assert_eq!(
+        cs.gates[17].verify_foreign_field_add::<Vesta>(0, &witness, &cs),
```

Comment:
You should use `gate.verify_witness::<G>(..)` instead. 

Also this means you can verify which constraint it failed on.  Also, it's way faster to run!

Reply:
Ah, nice then!